src/rules.c

```diff
@@ -711,11 +711,18 @@ char *rules_reject(char *rule, int split, char *last, struct db_main *db)
 			return NULL;
 
 		case 'u':
-			if (options.internal_cp == UTF_8 || options.internal_cp == ASCII) continue;
+			if (options.internal_cp == UTF_8 || options.internal_cp == ENC_RAW) continue;
 			return NULL;
 
 		case 'U':
-			if (options.internal_cp != UTF_8) continue;
+			if (options.internal_cp != UTF_8 && options.internal_cp != ENC_RAW) continue;
+			return NULL;
+
+/*
+ * Any failed UTF-8-to-codepage translations change '-U' to '--' in
+ * rpp_process_rule(), as an "always reject" hack.
+ */
+		case '-':
 			return NULL;
```

Comment:
@solardiz in my own view this is the only arguable part of this PR. I simply couldn't find a better way to handle a rule rejection that happens in `rpp_process_rule()` that would end up nicely.

Reply:
BTW I almost added this "-- always reject" to doc/RULES but decided it's of no interest to the user. We could add it anyway just for "documenting" it but OTOH as long as we don't we are free to change the flag to something else or even change this "hack" to something completely different and drop this new flag.

Reply:
I agree it's right to leave this hack undocumented.